conmon/conmon.c

```diff
@@ -319,8 +321,30 @@ int write_k8s_log(int fd, stdpipe_t pipe, const char *buf, ssize_t buflen)
 			}
 		}
 
+		write_len = line_len;
+
+		/*
+		 * We need to remove \r from a \r\n sequence while writing to the file.
+		 * This can be taken out once kubernetes enables OPOST in the client.
+		 */
+		if (*line_end == '\n' && line_len > 1) {
+			line_end--;
+			if (*line_end == '\r') {
+				buf[line_len - 2] = '\n';
```

Comment:
This modifies the passed in buffer, which is also sent to conn_sock. That doesn't seem right. At the very least it will cause conn_sock, to get two \n, but in general maybe we have the same CR/LF issues with the attaching fd?